apps/manifest-demo/3009-webpack-provider/webpack.config.js

```diff
@@ -19,6 +19,8 @@ module.exports = composePlugins(
     config.output.publicPath = 'auto';
     config.plugins.push(
       new ModuleFederationPlugin({
+        getPublicPath:
+          'function(path) {console.log("PUBLIC PATH", path); return path;}',
```

Comment:
@zhoushaw this is bit different from internal usage , but i think this is better to understand how to use getPublicPath . 

for our internal users, we can  auto transform the syntax to compat new implementation like this:

```js
// internal usage
const getPublicPath = `return "https:" + window.navigator.cdn_host + "/public_path/"`;

// transform

const replacedValue = `function(path) { ${getPublicPath} }`
 
```

Reply:
The reasoning here is to reduce the dependenance on new Function, this also allows us to pass arguments back from within the runtime to be calculated dynamically. For json remote we can pass the module, and original public path if hard coded. in normal use, we can pass webpack_require__.p which can remain set to "auto" and we can infer path off the calculated one. 

I can impement it as function body only and manually wrap string in webpack template to avoid new Function calls. 

new Function causes CSP issue in bank and other secure business environments. I try to ensure there is minimal use of new function when we can avoid it.

Reply:
We can also support both cases, i could check if it startsWith function() else wrap it in new function to support VMOK users? We can support backward compatibility so that there is no synchronization issue between older versions and newer versions?